### PR TITLE
Bugfix for trevc & many more LAPACK tests

### PR DESCRIPTION
**Pull request by kshyatt**
`trevc` wasn't accepting a valid input character. I added a lot of tests for untouched branches and error throws.

### CODE REVIEW

File: base/linalg/lapack.jl [hunk 1]
```diff
@@ -2262,7 +2262,9 @@ for (trcon, trevc, trrfs, elty) in
         function trevc!(side::Char, howmny::Char, select::Vector{BlasInt}, T::StridedMatrix{$elty},
                 VL::StridedMatrix{$elty} = similar(T), VR::StridedMatrix{$elty} = similar(T))
             # Extract
-            chkside(side)
+            if side ∉ ['L','R','B']
+                throw(ArgumentError(string("side argument must be 'L' (left eigenvectors), 'R' (right eigenvectors), or 'B' (both), got $side")))
```
Comment (tkelman) [inline base/linalg/lapack.jl]: is `string` redundant here?

Comment (kshyatt) [inline base/linalg/lapack.jl]: Probably.

**Comment by kshyatt** [comment]
Are the failures in push my fault?

**Comment by tkelman** [comment]
Probably not, ref 215fb9c90630ccdeb6f95d0f7b6cbbb5b724fb46 - a rebase that includes that reversion would likely make it go away

**Comment by kshyatt** [comment]
Do you want me to rebase and run CI again or is this ok to merge?

**Comment by andreasnoack** [comment]
LGTM

**Pull request merged**